### PR TITLE
Changes to Puppeteer

### PR DESCRIPTION
**Pull request by Knapptr**
phantom.js is deprecated

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+# Next.js build output
+.next
+out
+
+# Nuxt.js build / generate output
+.nuxt
+dist
+
+# Gatsby files
```
Comment (akahn) [inline .gitignore]: For the most part these ignores don't seem relevant to this project and should be put in users' own `~/.gitignore`. Please remove this file or only add files that need to be ignored that are specific to this project's setup.

File: README.md [hunk 2]
```diff
@@ -1,32 +1,28 @@
 # strava-heatmap-to-kmz
 
-Generate KMZ files from the [Strava Global Heatmap](http://labs.strava.com/heatmap/). This allows you to use the heatmap as a map layer (currently 1000×1000 pixels) on a mobile device (e.g. iPhone or Garmin), which enabling offline use and helps you to locate where the hell you are in the woods.
+Generate KMZ files from the [Strava Global Heatmap](http://labs.strava.com/heatmap/). This allows you to use the heatmap as a map layer (currently 1000×1000 pixels) on a mobile device (e.g. iPhone or Garmin), which enables offline use and helps you to locate where the hell you are in the woods.
```
Comment (akahn) [inline README.md:3]: 👍

File: main.js [hunk 3]
```diff
@@ -1,64 +1,95 @@
...
 
-var kmlTemplate = _.template(fs.read('kml.template', 'utf8'));
+	page.viewport({ width: 1000, height: 1000 });
+	await page.setCookie({
+		name: cookieKey,
+		value: sessionID,
+		domain: ".strava.com",
+	});
+	await page.goto(
+		`https://www.strava.com/heatmap#14.33/${lon}/${lat}/hot/ride`,
```
Comment (akahn) [inline main.js]: Can we go back to the gray color scheme?

**Review by akahn — CHANGES_REQUESTED** [review]
This is a great start! Thank you so much for doing this. I've requested a few small changes. Also, since we don't have any integration tests, could you post a screenshot showing the map loaded onto a Garmin device or app (I think GaiaGPS supports this, as does Google Earth).

**Comment by Knapptr** [comment]
Yup, can do, and will strip down that git ignore. I just generated one with
a one size fits all script 👍

File: main.js (hunk 3 above)
Comment (Knapptr) [reply main.js]: Yes

**Comment by Knapptr** [comment]
I defaulted behavior back to 'gray' but added the option of choice to colors. Readme updated
Stripped the .gitignore to be pertinent to the project

File: main.js [hunk 4]
```diff
@@ -1,64 +1,116 @@
...
-    east, west, north, south, bounds;
+//create template file
+const createTemplate = async () => {
+	const templateData = await fs.readFile("kml.template", "utf8");
+	const kmlTemplate = _.template(templateData);
+	return kmlTemplate;
+};
+
+//execution
+const reqUrl = `https://www.strava.com/heatmap#14.33/${lon}/${lat}/${colorString}/ride`;
```
Comment (akahn) [inline main.js]: I notice in the screenshot, there is some drift between the map base layer and the heatmap image on top. Is that because the zoom level changed from 15 to 14.33 in these changes?

**Comment by Knapptr** [comment]
Hm. You're right. I'll investigate

File: main.js (hunk 4 above)
Comment (Knapptr) [reply main.js]: It was not the zoom, but the resize event needed to be triggered after removal of dom elements. I returned to zoom #15, as its a nice integer, though.

**Comment by akahn** [comment]
This is great! Thanks for the patch.

**Review by akahn — APPROVED** [review]
